app.rb

```diff
@@ -1,20 +1,68 @@
 require_relative "data"
+require_relative "models/apartment"
+require_relative "models/tenant"
 apartments = data[:apartments]
 tenants = data[:tenants]
+ruby_apartments = []
+ruby_tenants = []
 
-# First, Open the data.rb an inspect the data. Identify and write, in comments, the following:
-  # Explain how the data is structured
-  # What are the properties for each of the two types of hashes
-
-# Use enumerables to -
-  # Print all the addresses for the apartments
-  # Print all the names for tenants
-  # Print only apartments that are less then 700 in rent
-  # Print only tenants that are over the age of 44
-  # Print only tenants that have an apartment id of 1
-  # Print all the tenants in order from youngest to oldest
-  # Print the names of all the tenants alphabetically
-
-  ## More challenging
-  # When printing tenants also print out the address that the tenant resides in.
-  # When printing all apartments, under each apartment print all of its tenants
+
+ruby_apartments = apartments.map do |apartment|
+  Apartment.new(apartment[:id], apartment[:address], apartment[:monthly_rent], apartment[:square_feet])
+end
+
+ruby_tenants = tenants.map do |tenant|
+  Tenant.new(tenant[:id], tenant[:name], tenant[:age], tenant[:apartment_id])
+end
+
+
+userInput = nil
+while userInput != "exit" do
+  puts "[Options Menu] \nType the number or phrase that corresponds with your choice."
+  puts "1: View all apartments"
+  puts "2: View all tenants"
+  puts "3: View all apartments with occupancy"
+  puts "4: View all tenants with address"
+  puts "Exit: Close program"
+  userInput = gets.chomp.downcase
+
+      if userInput == "1"
+        ruby_apartments.map do |apartment|
+          puts "Address: " + apartment.address
+          puts "Size: " + "#{apartment.square_feet}" + " sq ft"
+          puts "Monthly Rent: $" + "#{apartment.monthly_rent}"
+          puts "=" * 50
+        end
+
+      elsif userInput == "2"
+        ruby_tenants.map do |tenant|
+            puts "Name: " + tenant.name
+            puts "Age: " + "#{tenant.age}"
+            puts "=" * 50
+        end
+
+      elsif userInput == "3"
+        ruby_apartments.each do |apartment|
+          puts "=" * 50
+          puts apartment.address
+          puts "=" * 50
+          tenants_of_apartment = ruby_tenants.select{|tenant| tenant.apartment_id == apartment.id}
+          tenants_of_apartment.each do |tenant|
+            puts tenant.name
+          end
+          puts nil
+        end
+
+      elsif userInput == "4"
+        ruby_tenants.each do |tenant|
+          tenants_apartment_id = tenant.apartment_id
+          matching_apartments = ruby_apartments.find do |apartment|
+            apartment.id == tenants_apartment_id
+        end
+
+        puts "#{tenant.name} | #{matching_apartments.address}".center(50)
+        puts "=" * 50
+        end
+
+  end
```

Comment:
👍 !!! really awesome! One thing that could be helpful is to abstract these things into methods. In that way if you ever need to do it again, you can just call the method.